### DISCUSSION TITLE
Placing event on nodes before state plugin selects

### DISCUSSION THREAD

**Discussion by ghost** [vakata/jstree#2561] [Q&A] [answered]
Hello!

First of thanks for all the work on jstree it has helped me out a lot :).
But I have one small issue that I can't seem to figure out.

I want to put an event on the nodes for when they are clicked ('select_node.jstree') but I'm also using the state plugin.
So my idea is that I would first place the event for when a node is clicked and then the state plugin will select the node and my code in the event will run. 

Right now I'm adding the event when 'loaded.jstree' fires like so but it doesn't seem to be working consistently

```javascript
$('.jstree').on('loaded.jstree', function() {
    $('.jstree').on('select_node.jstree', function(event, node) {
        // ....
    });
});
```

I'm not sure if what I'm trying to do is possible but thanks in advance for any help :)

**Comment by vakata** [comment, marked as answer]
you can attach your event even before creating the tree instance itself - that way you won't miss any actions. You can also check if the selection was triggered by user interaction by inspecting the event's data argument - there should be an ```event``` property - if it is set - the selection was triggered by a user.

**Reply by ghost** [reply]
Makes sense, thank you!
